interface/main/main_screen.php

```diff
@@ -276,20 +319,33 @@ function generate_html_end()
             echo '    </div>';
             echo '</div>';
             if ($errormsg && $errortype == "U2F") {
-                echo '<div class="row"><div class="col-sm-12"><div class="alert alert-danger">' . text($errormsg) . '</div></div></div>';
+                echo '<div class="row"><div class="col-sm-12"><div class="alert alert-danger  alert-msg">' . text($errormsg) . '</div></div></div>';
             }
             echo '<div class="row">';
-            echo '    <div class="form-group">';
-            echo '        <label>' . xlt('Insert your key into a USB port and click the Authenticate button below.') . ' ' . xlt('Then press the flashing button on your key within 1 minute.') . '</label>';
-            echo '    </div>';
-            echo '    <div class="form-group">';
-            echo '        <a href="#" id="authutf" class="btn btn-default btn-save" onclick="doAuth()">' . xlt('Authenticate U2F') . '</a>';
-            echo '        <input type="hidden" name="form_requests" value="' . attr($requests) . '" />';
-            echo '        <input type="hidden" name="form_response" value="" />';
+            echo '  <div class="col-sm-12">';
+            echo '          <form method="post" name="u2fform" id="u2fform" action="main_screen.php?auth=login&site=' . attr_url($_GET['site']) . '" target="_top" name="challenge_form">';
+            echo '              <fieldset>';
+            echo '                  <legend>'. xlt('Insert U2F Key') .'</legend>';
+            echo '                  <div class="form-group">';
+            echo '                      <div class="col-sm-6 col-sm-offset-3">';
+            echo '                          <ul>';
+            echo '                              <li>' . xlt('Insert your key into a USB port and click the Authenticate button below.') . '</li>';
+            echo '                              <li>' . xlt('Then press the flashing button on your key within 1 minute.') . '</li>';
+            echo '                          </ul>';
+            echo '                  </div>';
+            echo '              </fieldset>';
+            echo '                  <div class="form-group clearfix">';
+            echo '                      <div class="col-sm-12 text-left position-override">';
+           //echo '                          <a href="#" id="authutf" class="btn btn-default btn-save" onclick="doAuth()">' . xlt('Authenticate U2F') . '</a>';
```

Comment:
better to not comment out code

Reply:
I could not test, due to lack of key

Reply:
That's fine. I'll test it out when next revision ready.

Reply:
testing well, can remove the line